napari/utils/colormaps/colormap.py

```diff
@@ -162,13 +162,28 @@ class LabelColormap(Colormap):
     interpolation: ColormapInterpolationMode = ColormapInterpolationMode.ZERO
     background_value: int = 0
 
-    def map(self, values) -> np.ndarray:
+    class Config:
+        keep_untouched = (cached_property,)
```

Comment:
This is a nested class? I find this a bit confusing...? Why does it exist? (it doesn't seem to be used anywhere else in this PR...?)

Reply:
This is the standard pydantic `Config` class to define properties for the evented model, and `keep_untouched` is one of the settings. https://docs.pydantic.dev/1.10/usage/model_config/

Reply:
This is required to make `cached_property` work. Otherwise it will fail on copy because of using threading lock. See: https://docs.python.org/3/library/functools.html#functools.cached_property 

It is required until python 3.12.

Reply:
Yuck. But anyway, since this is not standard Python, I suggest adding a comment pointing to both of those links.